mutis/correlation.py

```diff
@@ -75,14 +75,21 @@ def gen_synth(self, samples):
         self.signal1.gen_synth(samples)
         self.signal2.gen_synth(samples)
 
-    def gen_corr(self):
+    def gen_corr(self, uncert=True, dsamples=500):
         """Generates the correlation of the signals.
 
         Generates the correlation of the signals, and computes their
         confidence level from the synthetic light curves, which must
         have been generated before.
         """
 
+        if (uncert is True) and (self.signal1.dvalues is None):
+            log.error("uncert is True but no uncertainties for Signal 1 were specified")
+            uncert = False
+        if (uncert is True) and (self.signal2.dvalues is None):
+            log.error("uncert is True but no uncertainties for Signal 2 were specified")
+            uncert = False
+
         if not len(self.times) or not len(self.dts):
```

Comment:
do not use len(sequence) without comparison to determine if a sequence is empty..
`if not self.times or not self.dts`

Reply:
`Warning: The truth value of an empty array is ambiguous` so let's better do `if len(self.times) == 0 or len(self.dts) == 0`...

Reply:
In Python empty sequences evaluate `False` and non-empty evaluate `True`, so there's no need to use `len` with these objects. But, since these are not Python sequences/arrays/lists but `numpy` arrays, I guess it is better to use `np.array.size`, so let's just put

```
if self.times.size == 0 or self.dts.size == 0
```